pkg/errors/errors.go

```diff
@@ -7,9 +7,15 @@ import (
 
 // ErrorNotFound is returned when a 404 response is returned
 // from New Relic's APIs.
-type ErrorNotFound struct{}
+type ErrorNotFound struct {
+	Message string
```

Comment:
`ErrorUnexpectedStatusCode` uses `Err string` for this, not message.  Arguably should not be exported either, with a `New` and `Error()` functions the data within the struct can be hidden

Reply:
Good reference for custom errors in Go: https://golangbot.com/custom-errors/

Reply:
The terraform provider has [a need to check the error type](https://github.com/terraform-providers/terraform-provider-newrelic/pull/307/files#diff-39246a08dea1f12838e24b286d645fc8R219), so I think we'll need to export it.